lib/logstash/outputs/redis.rb

```diff
@@ -199,6 +229,67 @@ def connect
     Redis.new(params)
   end # def connect
 
+  def setup_ssl_params
+    require "openssl"
+
+    params = {}
+    params[:cert_store] = ssl_certificate_store
+
+    if @ssl_verification_mode == 'none'
+      params[:verify_mode] = OpenSSL::SSL::VERIFY_NONE
+    else
+      params[:verify_mode] = OpenSSL::SSL::VERIFY_PEER|OpenSSL::SSL::VERIFY_FAIL_IF_NO_PEER_CERT
+    end
+
+    if @ssl_certificate
+      params[:cert] = OpenSSL::X509::Certificate.new(File.read(@ssl_certificate))
+      if @ssl_key
+        # if we have an encrypted key and a password is not provided (nil) than OpenSSL::PKey::RSA
+        # prompts the user to enter a password interactively - we do not want to do that,
+        # for plain-text keys the default '' password argument gets simply ignored
+        params[:key] = OpenSSL::PKey::RSA.new(File.read(@ssl_key), @ssl_key_passphrase.value || '')
+      end
+    end
+
+    params[:min_version] = :TLS1_1
+    if @ssl_supported_protocols.any?
+      protocols = @ssl_supported_protocols.map { |v| "TLS#{v[-3..-1].gsub('.', '_')}".to_sym }.sort
```

Comment:
We can make the intent here a bit easier to read:
```suggestion
      protocols = @ssl_supported_protocols.map { |v| v.delete('v').tr(".", "_").to_sym }.sort
```